File: news-api/src/main/java/com/news/search/controller/FaviconController.java
package com.news.search.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class FaviconController {

    @RequestMapping("favicon.ico")
    public ResponseEntity<Void> favicon() {
        // favicon.ico 요청을 무시하고 204 No Content 상태를 반환
        return ResponseEntity.noContent().build();
    }
}

File: news-api/src/main/kotlin/com/search/config/favicon/FaviconController.kt
package com.search.config.favicon

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class FaviconController {
    @RequestMapping("favicon.ico")
    fun favicon(): ResponseEntity<Void> {
        return ResponseEntity.noContent().build()
    }
}